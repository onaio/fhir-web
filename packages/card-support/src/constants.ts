// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
