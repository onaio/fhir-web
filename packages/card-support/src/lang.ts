import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;
const lang: Lang = {};

/** recompute values */
function fill() {
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });
  lang.DOWNLOAD_CLIENT_DATA = i18n.t(`Download Client Data`, { ns: namespace });
  lang.CLIENT_LOCATION = i18n.t(`Client Location`, { ns: namespace });
  lang.ALL_LOCATIONS = i18n.t(`All Locations`, { ns: namespace });
  lang.CARD_STATUS = i18n.t(`Card Status`, { ns: namespace });
  lang.NEEDS_CARD = i18n.t(`Needs Card`, { ns: namespace });
  lang.CARD_NOT_NEEDED = i18n.t(`Card not needed`, { ns: namespace });
  lang.BOTH_NEEDS_CARD_CARD_NOT_NEEDED = i18n.t(`Both "Needs card" and "Card not needed"`, {
    ns: namespace,
  });
  lang.CARD_ORDER_DATE = i18n.t(`Card Order Date`, { ns: namespace });
  lang.CARD_ORDER_DATE_REQUIRED = i18n.t(`Please enter start date and end date`, { ns: namespace });
  lang.SELECT_CARD_ORDER_DATE = i18n.t(`Select Card Order Date`, { ns: namespace });
  lang.DOWNLOADING = i18n.t(`Downloading`, { ns: namespace });
  lang.DOWNLOAD_CSV = i18n.t(`Download CSV`, { ns: namespace });
  lang.NO_DATA_FOUND = i18n.t(`No data found`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
