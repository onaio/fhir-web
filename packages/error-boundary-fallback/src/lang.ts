import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

const lang: Dictionary<string> = {};

/** recompute values */
function fill() {
  lang.PAGE_TITLE = i18n.t(`An Error Occurred`, { ns: namespace });
  lang.PAGE_SUB_TITLE = i18n.t(
    `There has been an error. Itâ€™s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.'`,
    { ns: namespace }
  );
  lang.BUTTON_TITLE = i18n.t(`Back Home`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
