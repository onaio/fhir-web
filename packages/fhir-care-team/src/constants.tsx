// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
