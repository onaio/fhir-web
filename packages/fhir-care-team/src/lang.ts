import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;

const lang: Lang = {};

/** recompute values */
function fill() {
  // Tooltips
  lang.TIP_REQUIRED_FIELD = i18n.t(`This is a required field`, { ns: namespace });

  // Errors
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });

  // Messages
  lang.CARE_TEAMS_UPDATE_SUCCESS = i18n.t(`Successfully Updated Care Teams`, { ns: namespace });
  lang.CARE_TEAMS_ADD_SUCCESS = i18n.t(`Successfully Added Care Teams`, { ns: namespace });
  lang.CARE_TEAM_DELETE_SUCCESS = i18n.t(`Successfully Deleted Care Team`, { ns: namespace });
  lang.CONFIRM_DELETE = i18n.t(`Are you sure you want to delete this Care Team?`, {
    ns: namespace,
  });
  lang.YES = i18n.t(`Yes`, { ns: namespace });
  lang.NO = i18n.t(`No`, { ns: namespace });

  // Rendered text
  lang.STATUS = i18n.t(`Status`, { ns: namespace });
  lang.NAME = i18n.t(`Name`, { ns: namespace });
  lang.SAVING = i18n.t(`Saving`, { ns: namespace });
  lang.SAVE = i18n.t(`Save`, { ns: namespace });
  lang.ACTIVE = i18n.t(`Active`, { ns: namespace });
  lang.INACTIVE = i18n.t(`Inactive`, { ns: namespace });
  lang.CANCEL = i18n.t(`Cancel`, { ns: namespace });
  lang.EDIT_CARE_TEAM = i18n.t(`Edit Care Team`, { ns: namespace });
  lang.CREATE_CARE_TEAM = i18n.t(`Create Care Team`, { ns: namespace });
  lang.EDIT = i18n.t(`Edit`, { ns: namespace });
  lang.CREATE = i18n.t(`Create`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.TEAMS = i18n.t(`Care Teams`, { ns: namespace });
  lang.CARE_TEAM_MEMBERS = i18n.t(`Care Teams`, { ns: namespace });
  lang.SEARCH = i18n.t(`Search`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.CARE_TEAM_PAGE_HEADER = i18n.t(`FHIR Care Team`, { ns: namespace });
  lang.SUBJECT = i18n.t(`Subject`, { ns: namespace });
  lang.PARTICIPANTS = i18n.t(`Participant`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
