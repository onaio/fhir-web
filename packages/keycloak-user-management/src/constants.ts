// Web URLs
export const URL_ADMIN = '/admin';
export const URL_USER_EDIT = URL_ADMIN + '/users/edit';
export const URL_USER = URL_ADMIN + '/users/list';
export const URL_USER_GROUPS = `${URL_ADMIN}/users/groups`;
export const URL_USER_GROUP_EDIT = `${URL_ADMIN}/users/groups/edit`;
export const URL_USER_CREATE = URL_ADMIN + '/users/new';
export const URL_USER_GROUP_CREATE = `${URL_ADMIN}/users/groups/new`;
export const URL_USER_CREDENTIALS = URL_ADMIN + '/users/credentials';

// Route params
export const ROUTE_PARAM_USER_ID = 'userId';
export const ROUTE_PARAM_USER_GROUP_ID = 'userGroupId';

// Keycloak API URLs
export const KEYCLOAK_URL_USERS = '/users';
export const KEYCLOAK_URL_USER_GROUPS = '/groups';
export const KEYCLOAK_URL_RESET_PASSWORD = '/reset-password';
export const KEYCLOAK_URL_REQUIRED_USER_ACTIONS = '/authentication/required-actions/';

// OpenSRP API strings
export const OPENSRP_CREATE_PRACTITIONER_ENDPOINT = 'practitioner/user';

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';
