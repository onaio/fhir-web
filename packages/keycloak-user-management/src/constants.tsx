// Web URLs
export const URL_USER_EDIT = '/user/edit';
export const URL_ADMIN = '/admin';
export const URL_USER_CREATE = '/user/new';
export const URL_USER_CREDENTIALS = '/user/credentials';

// Route params
export const ROUTE_PARAM_USER_ID = 'userId';

// Keycloak API URLs
export const KEYCLOAK_URL_USERS = '/users';
export const KEYCLOAK_URL_RESET_PASSWORD = '/reset-password';
export const KEYCLOAK_URL_REQUIRED_USER_ACTIONS = '/authentication/required-actions/';

// Errors
export const ERROR_OCCURED = 'An error occurred';

// Messages
export const USER_DELETED_SUCCESSFULLY = 'User deleted successfully';
