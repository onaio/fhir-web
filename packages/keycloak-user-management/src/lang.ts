export const USER_MANAGEMENT_PAGE_HEADER = 'User Management';
export const ADD_USER = 'Add User';
