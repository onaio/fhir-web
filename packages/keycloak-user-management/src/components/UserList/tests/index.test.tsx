/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import reducerRegistry from '@onaio/redux-reducer-registry';
import fetch from 'jest-fetch-mock';
import * as fixtures from '../../forms/UserForm/tests/fixtures';
import { mount, shallow } from 'enzyme';
import { ConnectedUserList, UserList } from '..';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { act } from 'react-dom/test-utils';
import * as opensrpStore from '@opensrp/store';
import { Provider } from 'react-redux';
import { KeycloakService } from '@opensrp/keycloak-service';
import * as notifications from '@opensrp/notifications';
import {
  reducerName as keycloakUsersReducerName,
  reducer as keycloakUsersReducer,
  fetchKeycloakUsers,
  removeKeycloakUsers,
} from '../../../ducks/user';
import { keycloakUsersArray, keycloakUsersArray1 } from '../../forms/UserForm/tests/fixtures';
import { authenticateUser } from '@onaio/session-reducer';
import { URL_USER } from '../../../constants';
import lang from '../../../lang';

jest.mock('@opensrp/store', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/store')),
}));

jest.mock('@opensrp/notifications', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
}));

const history = createBrowserHistory();

const locationProps = {
  history,
  location: {
    hash: '',
    pathname: `${URL_USER}`,
    search: '',
    state: {},
  },
  match: {
    isExact: true,
    params: {},
    path: `${URL_USER}`,
    url: `${URL_USER}`,
  },
};

reducerRegistry.register(keycloakUsersReducerName, keycloakUsersReducer);

describe('components/UserList', () => {
  beforeEach(() => {
    fetch.resetMocks();
    opensrpStore.store.dispatch(removeKeycloakUsers());
  });

  beforeAll(() => {
    opensrpStore.store.dispatch(
      authenticateUser(
        true,
        {
          email: 'bob@example.com',
          name: 'Bobbie',
          username: 'RobertBaratheon',
        },
        // eslint-disable-next-line @typescript-eslint/camelcase
        { api_token: 'hunter2', oAuth2Data: { access_token: 'simple-token', state: 'abcde' } }
      )
    );
  });

  it('renders users table without crashing', () => {
    shallow(<UserList {...locationProps} />);
  });
  it('works correctly with store', async () => {
    fetch.once(JSON.stringify(4));
    fetch.once(JSON.stringify(fixtures.keycloakUsersArray));
    const props = {
      ...locationProps,
      extraData: {
        user_id: fixtures.keycloakUser.id,
      },
      fetchKeycloakUsersCreator: fetchKeycloakUsers,
      removeKeycloakUsersCreator: removeKeycloakUsers,
      serviceClass: KeycloakService,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
      usersPageSize: 20,
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <ConnectedUserList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    expect(wrapper.find('Table')).toBeTruthy();
    expect(wrapper.text()).toMatchSnapshot('full rendered text');
    wrapper.unmount();
  });

  it('renders user list correctly', async () => {
    fetch.once(JSON.stringify(4));
    fetch.once(JSON.stringify(keycloakUsersArray));
    const props = {
      ...locationProps,
      extraData: {
        user_id: fixtures.keycloakUser.id,
      },
      fetchKeycloakUsersCreator: fetchKeycloakUsers,
      removeKeycloakUsersCreator: removeKeycloakUsers,
      serviceClass: KeycloakService,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
      usersPageSize: 20,
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <ConnectedUserList {...props} />
        </Router>
      </Provider>
    );
    // Loader should be displayed
    expect(toJson(wrapper.find('.ant-spin'))).toBeTruthy();

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    // Loader should be hiddern
    expect(toJson(wrapper.find('.ant-spin'))).toBeFalsy();

    expect(fetch.mock.calls).toEqual([
      [
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage/users/count',
        {
          headers: {
            accept: 'application/json',
            authorization: 'Bearer simple-token',
            'content-type': 'application/json;charset=UTF-8',
          },
          method: 'GET',
        },
      ],
      [
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage/users?first=0&max=20',
        {
          headers: {
            accept: 'application/json',
            authorization: 'Bearer simple-token',
            'content-type': 'application/json;charset=UTF-8',
          },
          method: 'GET',
        },
      ],
    ]);

    const userList = wrapper.find('UserList');
    const headerRow = userList.find('Row').at(0);

    expect(headerRow.find('Col').at(0).text()).toMatchSnapshot('header actions col props');
    expect(headerRow.find('Table').first().text()).toMatchSnapshot('table text');
    // look for pagination
    expect(wrapper.find('Pagination').at(0).text()).toMatchInlineSnapshot(`"120 / page"`);
    wrapper.unmount();
  });

  it('search works correctly', async () => {
    fetch.once(JSON.stringify(4));
    fetch.once(JSON.stringify(keycloakUsersArray));
    const props = {
      ...locationProps,
      location: {
        ...locationProps.location,
        search: '?searchQuery=opensrp',
      },
      extraData: {
        user_id: fixtures.keycloakUser.id,
      },
      fetchKeycloakUsersCreator: fetchKeycloakUsers,
      removeKeycloakUsersCreator: removeKeycloakUsers,
      serviceClass: KeycloakService,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
      usersPageSize: 20,
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <ConnectedUserList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    const search = wrapper.find('.search-input-wrapper');
    // fireEvent.change(search, { target: { value: 'opensrp' } });
    await act(async () => {
      search.find('.ant-input').simulate('change', { target: { value: 'test' } });
    });
    await flushPromises();
    wrapper.update();
    expect(fetch.mock.calls).toEqual([
      [
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage/users/count',
        {
          headers: {
            accept: 'application/json',
            authorization: 'Bearer simple-token',
            'content-type': 'application/json;charset=UTF-8',
          },
          method: 'GET',
        },
      ],
      [
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage/users?first=0&max=20&search=opensrp',
        {
          headers: {
            accept: 'application/json',
            authorization: 'Bearer simple-token',
            'content-type': 'application/json;charset=UTF-8',
          },
          method: 'GET',
        },
      ],
    ]);
  });

  it('pagination works', async () => {
    fetch.once(JSON.stringify(7));
    fetch.once(JSON.stringify([...keycloakUsersArray, ...keycloakUsersArray1]));
    const props = {
      ...locationProps,
      extraData: {
        user_id: fixtures.keycloakUser.id,
      },
      fetchKeycloakUsersCreator: fetchKeycloakUsers,
      removeKeycloakUsersCreator: removeKeycloakUsers,
      serviceClass: KeycloakService,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
      usersPageSize: 5,
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <ConnectedUserList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    wrapper.find('.ant-pagination-item-2').simulate('click');
    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"User ManagementAdd UserUsernameEmailFirst NameLast NameActionszkaliminaZembaKaliminaEditzkapeleZyingaKapeleEdit125 / pageGo to)"`
    );
  });

  it('redirects to new user form', async () => {
    const historyPushMock = jest.spyOn(history, 'push');
    fetch.once(JSON.stringify(4));
    fetch.once(JSON.stringify(fixtures.keycloakUsersArray));
    const props = {
      ...locationProps,
      extraData: {
        user_id: fixtures.keycloakUser.id,
      },
      fetchKeycloakUsersCreator: fetchKeycloakUsers,
      removeKeycloakUsersCreator: removeKeycloakUsers,
      serviceClass: KeycloakService,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
      usersPageSize: 20,
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <ConnectedUserList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    // expect(toJson(wrapper.find('.create-user').at(1))).toEqual('');
    wrapper.find('.create-user').at(0).simulate('click');
    expect(historyPushMock).toHaveBeenCalledWith('/admin/users/new');
  });

  it('handles user list fetch failure', async () => {
    fetch.mockReject(() => Promise.reject('API is down'));
    fetch.mockReject(() => Promise.reject('API is down'));
    const mockNotificationError = jest.spyOn(notifications, 'sendErrorNotification');
    const props = {
      ...locationProps,
      extraData: {
        user_id: fixtures.keycloakUser.id,
      },
      fetchKeycloakUsersCreator: fetchKeycloakUsers,
      removeKeycloakUsersCreator: removeKeycloakUsers,
      serviceClass: KeycloakService,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <ConnectedUserList {...props} />
        </Router>
      </Provider>
    );

    // Loader should be displayed
    expect(toJson(wrapper.find('.ant-spin'))).toBeTruthy();

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    /**
     * Loader should not be displayed
     * since we've set loading to false
     * on the final block
     */
    expect(toJson(wrapper.find('div.lds-ripple'))).toBeFalsy();
    // check that table has No Data
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"User ManagementAdd UserUsernameEmailFirst NameLast NameActionsNo Data)"`
    );
    expect(mockNotificationError).toHaveBeenCalledWith(lang.ERROR_OCCURED);
  });
});
