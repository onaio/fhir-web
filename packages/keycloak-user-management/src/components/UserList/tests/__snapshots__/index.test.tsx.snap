// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UserList renders user list correctly: header actions col props 1`] = `"Add UserUsernameEmailFirst NameLast NameActionsmwalimudubdabasoduba@gmail.comBenjaminMwalimuEditonaOnakenyaEditona-admintest@onatest.comOnaAdminEditopensrpDemokenyaEdit120 / page)"`;

exports[`components/UserList renders user list correctly: table text 1`] = `"UsernameEmailFirst NameLast NameActionsmwalimudubdabasoduba@gmail.comBenjaminMwalimuEditonaOnakenyaEditona-admintest@onatest.comOnaAdminEditopensrpDemokenyaEdit120 / page"`;

exports[`components/UserList sort works correctly: sorted table rows by email - ascending 0 1`] = `"UsernameEmailFirst NameLast NameActions"`;

exports[`components/UserList sort works correctly: sorted table rows by email - ascending 1 1`] = `"mwalimudubdabasoduba@gmail.comBenjaminMwalimuEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - ascending 2 1`] = `"onaOnakenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - ascending 3 1`] = `"ona-admintest@onatest.comOnaAdminEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - ascending 4 1`] = `"opensrpDemokenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - descending 0 1`] = `"UsernameEmailFirst NameLast NameActions"`;

exports[`components/UserList sort works correctly: sorted table rows by email - descending 1 1`] = `"mwalimudubdabasoduba@gmail.comBenjaminMwalimuEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - descending 2 1`] = `"onaOnakenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - descending 3 1`] = `"ona-admintest@onatest.comOnaAdminEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by email - descending 4 1`] = `"opensrpDemokenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - ascending 0 1`] = `"UsernameEmailFirst NameLast NameActions"`;

exports[`components/UserList sort works correctly: sorted table rows by username - ascending 1 1`] = `"opensrpDemokenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - ascending 2 1`] = `"ona-admintest@onatest.comOnaAdminEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - ascending 3 1`] = `"onaOnakenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - ascending 4 1`] = `"mwalimudubdabasoduba@gmail.comBenjaminMwalimuEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - descending 0 1`] = `"UsernameEmailFirst NameLast NameActions"`;

exports[`components/UserList sort works correctly: sorted table rows by username - descending 1 1`] = `"mwalimudubdabasoduba@gmail.comBenjaminMwalimuEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - descending 2 1`] = `"onaOnakenyaEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - descending 3 1`] = `"ona-admintest@onatest.comOnaAdminEdit"`;

exports[`components/UserList sort works correctly: sorted table rows by username - descending 4 1`] = `"opensrpDemokenyaEdit"`;

exports[`components/UserList sort works correctly: table rows - default order 0 1`] = `"UsernameEmailFirst NameLast NameActions"`;

exports[`components/UserList sort works correctly: table rows - default order 1 1`] = `"mwalimudubdabasoduba@gmail.comBenjaminMwalimuEdit"`;

exports[`components/UserList sort works correctly: table rows - default order 2 1`] = `"onaOnakenyaEdit"`;

exports[`components/UserList sort works correctly: table rows - default order 3 1`] = `"ona-admintest@onatest.comOnaAdminEdit"`;

exports[`components/UserList sort works correctly: table rows - default order 4 1`] = `"opensrpDemokenyaEdit"`;

exports[`components/UserList works correctly with store: full rendered text 1`] = `"User ManagementAdd UserUsernameEmailFirst NameLast NameActionsmwalimudubdabasoduba@gmail.comBenjaminMwalimuEditonaOnakenyaEditona-admintest@onatest.comOnaAdminEditopensrpDemokenyaEdit120 / page)"`;
